tests/rsapss.cc

```diff
@@ -0,0 +1,214 @@
+                                             e.data(),
+                                             saltLen,
+                                             sgnt.size(),
+                                             sgnt.data(),
+                                             msg.size(),
+                                             msg.data());
+
+  ASSERT_EQ(got1, got2) << "`Hacl_RSAPSS_rsapss_verify(...)` deviates from "
+                           "`Hacl_RSAPSS_rsapss_pkey_verify(...)`.";
+
+  *out = got2;
+}
+
+class RsaPssVerifySuite : public ::testing::TestWithParam<Group>
+{};
+
+TEST_P(RsaPssVerifySuite, Group)
+{
+  auto group = GetParam();
+
+  // std::cout << "e: " << bytes_to_hex(group.e) << std::endl
+  //           << "n: " << bytes_to_hex(group.n) << std::endl;
+
+  for (auto test : group.tests) {
+    // std::cout << "msg: " << bytes_to_hex(test.msg) << std::endl
+    //           << "sgnt: " << bytes_to_hex(test.sgnt) << std::endl;
+
+    bool got;
+    verify(group.e, group.n, group.saltLen, group.a, test.msg, test.sgnt, &got);
+
+    EXPECT_EQ(test.expected, got) << "tcId=" << test.tcId;
+  }
+}
+
+std::vector<Group>
+read_json(char* path)
+{
+  json tests;
+  std::ifstream file(path);
+  file >> tests;
+
+  std::vector<Group> testGroups;
+
+  for (auto& group_raw : tests["testGroups"].items()) {
+    auto group = group_raw.value();
+
+    bytes e = from_hex(group["e"]);
+    bytes n = from_hex(group["n"]);
+
+    // Remove first 0x00 byte in n.
+    if (n[0] == 0x00) {
+      n.erase(n.begin());
+    } else {
+      std::ostringstream msg;
+      msg << "Expected first byte of \"n\" to be 0x00 in Wycheproof test (path=" << path << ").";
+
+      throw std::invalid_argument(msg.str());
+    }
+
+    uint32_t saltLen = group["sLen"];
+
+    Spec_Hash_Definitions_hash_alg a;
+    std::string sha = group["sha"];
+    if (sha == "SHA-256") {
+      a = Spec_Hash_Definitions_SHA2_256;
+    } else if (sha == "SHA-512") {
+      a = Spec_Hash_Definitions_SHA2_512;
+    } else {
+      std::ostringstream msg;
+      msg << "Unexpected value \"" << sha
+          << "\" in field \"sha\" (path=" << path << ").";
+
+      throw std::invalid_argument(msg.str());
+    }
+
+    std::vector<Test> tests;
+    for (auto& test_raw : group["tests"].items()) {
+      auto test = test_raw.value();
+      uint32_t tcId = test["tcId"];
+
+      auto msg = from_hex(test["msg"]);
+      auto sgnt = from_hex(test["sig"]);
+
+      bool expected;
+      std::string result = test["result"];
+      if (result == "valid" || result == "acceptable") {
+        expected = true;
+      } else if (result == "invalid") {
+        expected = false;
+      } else {
+        std::ostringstream msg;
+        msg << "Unexpected value \"" << result
+            << "\" in field \"result\" (file=" << path << ", tcId=" << tcId
+            << ").";
+
+        throw std::invalid_argument(msg.str());
+      }
+
+      tests.push_back(Test{
+        tcId : tcId,
```

Comment:
The compiler complains about this old style.
```suggestion
        .tcId = tcId,
```